pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/signature/JavaOperationSignature.java

```diff
@@ -64,10 +66,10 @@ private static int code(Visibility visibility, Role role, boolean isAbstract) {
      *
      * @return The signature of the parameter
      */
-    public static OperationSignature buildFor(ASTMethodOrConstructorDeclaration node) {
+    public static JavaOperationSignature buildFor(ASTMethodOrConstructorDeclaration node) {
         int code = code(Visibility.get(node), Role.get(node), node.isAbstract());
```

Comment:
Similar like the comment for JavaFieldSignature: we have a limited number of combinations (visibility, role, abstract) - so, I guess, we won't run into a collision...